### DISCUSSION TITLE
Waze map integration

### DISCUSSION THREAD

**Discussion by Vladutu** [BimmerGestalt/AAIdrive#769] [Q&A] [answered]
Hi! First i would like to thank you for the application. It's a game changer for us android users :). I saw that waze has a transport sdk and from what I saw it's free. I wanted to ask if it can be implemented in the same way as google maps/mapbox.
Thank you!

**Comment by hufman** [comment, marked as answer]
The [Waze Transport SDK](https://developers.google.com/waze/intro-transport) doesn't seem feasible for a few reasons.
It requires having an existing partner relationship, and they are unlikely to partner with a small-time opensource app developer. It also specifically says that "The SDK is *not* for partners to build their own navigation app", and it does not provide "The Waze map and navigation embedded in your app".
Given the benefit of letting "Drivers seamlessly switch between your app and Waze to find the fastest routes", it's meant for delivery drivers to use Waze as the navigation app, and to have a button in the delivery app to quickly open Waze, while sharing ETA information with the delivery app.

What SDK did you find? What features do you wish for from Waze?

**Reply by Vladutu** [reply]
I found the same sdk, i saw the "The SDK is not for partners to build their own navigation app" but didn't know for sure it applies or not to the application that you did. 
What I wanted to have is a fully integrated waze (same as you did with google maps) but also having voice directions. I thought that maybe it would be possible to use waze directly. 
Thank you for the answer!

**Reply by hufman** [reply]
Indeed, it looks like this SDK does not provide the Waze map to be embedded in another app, it just provides a quick way to open Waze as a separate app.
AAIdrive does not have an integrated Google Maps feature. It uses the Android Mapview (which somewhat looks like Google Maps), and it uses the Maps API for place search and a route lookup. It can't provide features that the real Google Maps provides, such as listing your favorite places from your account, and it is not allowed to implemented the turn-by-turn navigation experience.
It is impossible to embed another app into a hosting app for easy screenshotting and control without support from the other app, which is unlikely to happen for any of these big navigation apps.
